p2p/listener.go

```diff
@@ -304,60 +293,23 @@ func (m *listener) providerStartConfigExchange(providerID identity.Identity, msg
 // required ports count for actual p2p and service connections and fallback to
 // acquiring extra ports for nat pinger if provider is behind nat, port mapping failed
 // and no manual port forwarding is enabled.
-func (m *listener) prepareLocalPorts(id, outboundIP string, tracer *trace.Tracer) (string, []int, []func(), error) {
+func (m *listener) prepareLocalPorts(id string, tracer *trace.Tracer) (string, []int, []func(), nat.StartPorts, error) {
```

Comment:
At some point the returns values here should be remade. Adding one more return here is just more confusion.

Reply:
It could be wrapped into struct with meaningful names attached

Reply:
I'm going to change it with the next couple of PRs. So merging like this now.